src/psyclone/tests/dependency_test.py

```diff
@@ -505,3 +509,403 @@ def test_math_equal_limitations(parser):
+    assert "op1_proxy%ncell_3d: READ" in var_info
+
+
+def test_lfric_cma2():
+    '''Test that parameters related to CMA operators are handled
+    correctly in the variable usage analysis.
+
+    '''
+    _, invoke_info = get_invoke("20.1_cma_apply.f90", "dynamo0.3", idx=0)
+    var_info = str(VariablesAccessInfo(invoke_info.schedule))
+    assert "cma_indirection_map_aspc1_field_a: READ" in var_info
+    assert "cma_indirection_map_aspc2_field_b: READ" in var_info
+
+
+def test_lfric_stencils():
+    '''Test that stencil parameters are correctly detected.
+
+    '''
+    _, invoke_info = get_invoke("14.4_halo_vector.f90", "dynamo0.3", idx=0)
+    var_info = str(VariablesAccessInfo(invoke_info.schedule))
+    assert "f2_stencil_size: READ" in var_info
+    assert "f2_stencil_dofmap: READ" in var_info
+
+
+def test_lfric_operator_different_spaces():
+    ''' Tests that implicit parameters for an operator with different to
+    and from spaces are created correctly.
+
+    '''
+    _, invoke_info = get_invoke("10.3_operator_different_spaces.f90",
+                                "dynamo0.3", idx=0)
+    var_info = str(VariablesAccessInfo(invoke_info.schedule))
+    assert "orientation_w2: READ" in var_info
+
+
+def test_lfric_field_bc_kernel():
+    '''Tests that implicit parameters in case of a boundary_dofs
+    array fix are created correctly.
+
+    '''
+    _, invoke_info = get_invoke("12.2_enforce_bc_kernel.f90",
+                                "dynamo0.3", idx=0)
+    var_info = str(VariablesAccessInfo(invoke_info.schedule))
+    assert "boundary_dofs_a: READ" in var_info
+
+
+def test_lfric_stencil_xory_vector():
+    '''Test that the implicit parameters for a stencil access of type x
+    or y with a vector field are created.
+
+    '''
+    _, invoke_info = get_invoke("14.4.2_halo_vector_xory.f90",
+                                "dynamo0.3", idx=0)
+    var_info = str(VariablesAccessInfo(invoke_info.schedule))
+    assert "f2_direction: READ" in var_info
+
+
+def test_lfric_operator_bc_kernel():
+    '''Tests that a kernel that applies boundary conditions to operators
+    detects the right implicit paramaters.
+
+    '''
+    _, invoke_info = get_invoke("12.4_enforce_op_bc_kernel.f90",
+                                "dynamo0.3", idx=0)
+    var_info = str(VariablesAccessInfo(invoke_info.schedule))
+    assert "boundary_dofs_op_a: READ" in var_info
+
+
+def test_lfric_stub_args():
+    '''Check that correct stub code is produced when there are multiple
+    stencils.
+
+    '''
+    from psyclone.dynamo0p3 import DynKernMetadata, DynKern
+    from psyclone.domain.lfric import KernStubArgList
+    ast = get_ast("dynamo0.3", "testkern_stencil_multi_mod.f90")
+    metadata = DynKernMetadata(ast)
+    kernel = DynKern()
+    kernel.load_meta(metadata)
+    var_accesses = VariablesAccessInfo()
+    create_arg_list = KernStubArgList(kernel)
+    create_arg_list.generate(var_accesses=var_accesses)
+
+
+def test_lfric_stub_args2():
```

Comment:
Very good question. I admit initially I just left it out, but the obvious effect is many pylint warnings  because of inconsistencies in the name of abstract functions in the base class (which of course could be disabled). So I decided to make it all consistent, based on "Who knows when this might be useful, I certainly don't want to get back to this in three months time" ... but I am certainly equally happy to remove all of it, I couldn't think of a realistic, _immediate_ need (and was more thinking that the code reviewer would not like the inconsistencies).

Reply:
OK, still we make a different decision, I'll leave it in.